metrics/Cumulative.md

```diff
@@ -0,0 +1,54 @@
+	* `labelValues`: the list of label values. The number of label values must be the same to that of the label keys.
+* The `Point` class should provide functionalities to manually increment/reset values. Example: `inc()` bye one, `inc(long times)` a specific number of times at once, `reset()`.
+* Remove a single time series from the cumulative metric, if it is present.
+	* `labelValues`: the list of label values.
+* Clear all time series from the cumulative metric i.e. References to all previous point objects are invalid (not part of the metric).
+* Get a default time series for a cumulative with all labels not set, or default labels.
+
+
+It is recommended to keep a reference of a point for manual operations instead of always calling `getOrCreateTimeSeries` method. The keys in `labelKeys` must not be present in `constantLabels` map. Also, `constantLabels` will be added to all the timeseries for the Metric.
+
+## Derived Cumulative API
+
+The value that is published for cumulative is an instantaneous measurement of an `int64` or `double` value. This cumulative is self sufficient once created, so users should never need to interact with it. The value of the cumulative is observed from the `object` and a `callback function`. The callback function is invoked whenever metrics are collected, meaning the reported value is up-to-date.
+
+The following general operations MUST be provided by the API:
+
+* Defining a `name`, `description`, `unit`, `labelKeys`, `resource` and `constantLabels` which are fixed labels that always apply to a cumulative. This should give back cumulative object to add new time series, remove time series and clear all time series.
+	* `name`: same as above name.
+	* `description`: same as above description.
+	* `unit`: same as above unit.
+	* `labelKeys`: same as above labelKeys.
+	* `constantLabels`: same as above constantLabels.
+	* `resource`: same as above resource.
+* Add a new time series with label values, an `object` and a `callback function`. The number of label values must be the same to that of the label keys.
+	* `labelValues`: the list of label values. The number of label values must be the same to that of the label keys.
+	* `object`: the state object from which the function derives a measurement.
+	* `function`: the callback function to be called.
+* Remove a single time series from the cumulative metric, if it is present.
+	* `labelValues`: the list of label values.
+* Clear all time series from the cumulative metric i.e. References to all previous point objects are invalid (not part of the metric).
+
+In a time series, cumulative metric should have the same start time (set when creating a cumulative) and increasing end times, until an event resets then the start time should also be reset and we should use new start time for the following points.
```

Comment:
What if the function registered returns a value that is lower than before (due to reset or some other reason) without application calling reset? Do we
1. reset the start time whenever we detect that new value is not higher than the old value. OR
2. continue to report old value.

I think we should do (1).